Implements during ReQL command

zedtux opened the pull request:
This PR implements [the `during` ReQL command](https://rethinkdb.com/api/ruby/during/) which allow to filter documents having the given filed between the 2 given times.

Here is an example of its usage:

```ruby
User.where(:created_at.during(1.month.ago, Time.current)).to_a
```

It returns all the created `User`s in the last month period of time.

It replaces using the combinaison of `gt` and `lt`.

zedtux commented:
There's an issue with type casting so please wait before to more this one. Sorry about that.

nviennot commented on spec/integration/criteria/where_spec.rb:
```diff
@@ -28,7 +28,7 @@
   context 'when using bad arguments' do
     it 'raises' do
       expect { SimpleDocument.where("xxx") }.to raise_error(/Invalid/)
-      expect { SimpleDocument.where(:xxx.eq(1,2)) }.to raise_error(/Invalid/)
```
eq(1,2) should still raise an error

zedtux replied:
Nice catch! Of course. I'll change that, thank you!

zedtux commented:
@nviennot I fixed your review and added the TimeWithZone support for `during`.

When you'll have accepted it, I can submit a PR to update the doc if you'd like to 😉
